components/team_general_tab/team_general_tab.jsx

```diff
@@ -823,7 +708,20 @@ export default class GeneralTab extends React.PureComponent {
                         </>
                     }
                     <div className='divider-light'/>
-                    {openInviteSection}
+                    <OpenInvite
+                        teamId={this.props.team.id}
+                        isActive={this.props.activeSection === 'open_invite'}
+                        isGroupConstrained={this.props.team.group_constrained}
+                        allowOpenInvite={this.props.team.allow_open_invite}
+                        onToggle={(active) => {
```

Comment:
Right now this is fine, but if you plan to extract more of those parts it would probably a good idea to have some kind of generator function, like this:

```typescript jsx
const getOnToggle = (section) => (active) => this.handleUpdateSection(active ? section : '');
```

and then use it like this: 

```typescript jsx
...
onToggle={getOnToggle('open_invite')}
...
```

Reply:
Note that both the current way we're doing it and the proposed way generate new functions and cause the OpenInvite to re-render whenever the TeamGeneralTab does. I don't think this code is exactly performance-sensitive though, and we can always memoize that with `useCallback` once all of this is made into a functional component though

Reply:
You can also wrap the component into `React.memo` to prevent unnecessary re-renders. But the proposed way of adding `useCallback` should do the trick as well

Reply:
`React.memo` on its own won't fix the case where the `TeamGeneralTab` re-rendering causes the `OpenInvite` to re-render since it's still receiving a new version of `onToggle` each time

Reply:
@hmhealey you are absolutely right.